lib/addons/p5.dom.js

```diff
@@ -675,6 +675,84 @@
   };
 
   /**
+   * Translates an element relative to (0, 0, 0) of the window through
+   * css transforms. 
+   * @method  translate
+   * @param  {Number} x x-position relative to upper left of window
+   * @param  {Number} y y-position relative to upper left of window
+   * @param  {Number} z z-position
+   * @return {p5.Element}
+   * @example
+   * <div><code class='norender'>
+   * function setup() {
+   *   var cnv = createCanvas(100,100);
+   *   //translates canvas 50px to the right and 100px
+   *   //below upper left corner of the window
+   *   cnv.translate(50,100);
+   * }
+   * </code></div>
+   */
+  p5.Element.prototype.translate = function(x, y, z){
+    this.elt.style.position = 'absolute';
+    var regEx = new RegExp('translate');
+    var substr = '';
+    while (regEx.test(this.elt.style.transform)){
+      var min = this.elt.style.transform.indexOf('translate');
+      var max = this.elt.style.transform.indexOf(')');
+      if (min != -1) {
+        substr = this.elt.style.transform.substring(min, max+1);
+        this.elt.style.transform = this.elt.style.transform.replace(substr, '');
+      }
+    }
+    this.elt.style.transform += 'translate3d('+x+'px,'+y+'px,'+z+'px)';
+    return this;
+  };
+
+  /**
+   * Rotates an element through css transforms in 3 dimensional space.
+   * @method  rotate
+   * @param  {Number} x amount of degrees to rotate the element along the x-axis
+   * @param  {Number} y amount of degrees to rotate the element along the y-axis
+   * @param  {Number} z amount of degrees to rotate the element along the z-axis
+   * @return {p5.Element}
+   * @example
+   * <div><code class='norender'>
+   * var cnv,
+   *     x = 0,
+   *     y = 0,
+   *     z = 0;
+   * 
+   * function setup(){
+   *   cnv = createCanvas(100,100);
+   * }
+   *
+   * function draw(){
+   *   //rotates the canvas .5deg (degrees) on every axis each frame.
+   *   x+=.5 % 360;
+   *   y+=.5 % 360;
+   *   z+=.5 % 360;
+   *   cnv.rotate(x,y,z);
+   * }
+   * </code></div>
+   */
+  p5.Element.prototype.rotate = function(x, y, z){
+    var regEx = new RegExp('rotate');
+    var substr = '';
+    while (regEx.test(this.elt.style.transform)){
+      var min = this.elt.style.transform.indexOf('rotate');
+      var max = this.elt.style.transform.indexOf(')');
+      if (min != -1) {
+          substr = this.elt.style.transform.substring(min, max+1);
+          this.elt.style.transform = this.elt.style.transform.replace(substr, '');
+      }
+    }
+    this.elt.style.transform += 'rotateX('+x+'deg)';
+    this.elt.style.transform += 'rotateY('+y+'deg)';
+    this.elt.style.transform += 'rotateZ('+z+'deg)';
```

Comment:
same thing here, this should probably accept either 1 or 3 args, and use `rotate()` or `rotateX()` etc. this can also be indicated in the documentation above by putting [] around the optional args.